model/registry/cloudfoundry.yaml

```diff
@@ -0,0 +1,140 @@
+groups:
+  - id: registry.cloudfoundry
+    prefix: cloudfoundry
+    type: resource
+    brief: >
+      CloudFoundry resource attributes.
+    attributes:
+      - id: source.id
+        type: string
+        brief: >
+          A guid or another name describing the event source.
+        note: |
+          CloudFoundry defines the `source_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the application id for applications deployed on
+          the runtime or the component name, e.g. "gorouter", for CloudFoundry
+          components.
+
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_id`. This is the same value as
+          reported by `cf app <app-name> --guid`.
+
+          When system components are instrumented, values from the [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `source.id` should be set to
+          `spec.deployment/spec.name`.
+        examples:
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+          - 'cf/gorouter'
+      - id: instance.id
+        type: string
+        brief: >
+          An index or guid describing the concrete instance of the event source.
+        note: |
+          CloudFoundry defines the `instance_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the application instance index for applications
+          deployed on the runtime or the vm id for CloudFoundry components.
+
+          Application instrumentation should use the value from environment
+          variable `CF_INSTANCE_INDEX`.
+
+          When system components are instrumented, values from the [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `instance.id` should be set to `spec.id`.
+        examples:
+          - '0'
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+      - id: app.name
+        type: string
+        brief: >
+          The name of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_name`. This is the same value
+          as reported by `cf apps`.
+        examples: ['my-app-name']
+      - id: app.id
```

Comment:
Please have a look at this PR https://github.com/open-telemetry/semantic-conventions/pull/312. It seems this is exactly the same there.

Reply:
`app.id` is different than `instance.id`. The former is the GUID of the app that the signal originates from, and the latter refers to the VM that the app or system component is running on.

Reply:
@ctlong to my understanding of https://github.com/cloudfoundry/loggregator-api#v2-envelope, the instance id will be the app instance index and not a cell or vm id for applications. The GUID is only used for system components such as the Gorouter. This is also my experience with these kind of signals.